lib/rust/ensogl/core/src/debug/stats.rs

```diff
@@ -103,3 +247,43 @@ macro_rules! if_compiled_with_stats {
         {}
     };
 }
+
+
+
+// ========================
+// === ValueAccumulator ===
+// ========================
+
+#[derive(Debug, Default)]
+struct ValueAccumulator<T> {
+    pub min: T,
+    pub max: T,
+    pub sum: f64,
+}
+
+impl<T: Min + Max + PartialOrd + cast::AsPrimitive<f64> + Copy> ValueAccumulator<T> {
+    fn new(v: T) -> Self {
+        Self { min: v, max: v, sum: v.as_() }
+    }
+
+    fn push(&mut self, v: T) {
+        self.min = min(self.min, v);
+        self.max = max(self.max, v);
+        self.sum += v.as_();
+    }
+}
+
+
+
+// ====================
+// === ValueSummary ===
+// ====================
+
+/// Summarized data observed for a single stat over a number of GUI rendering frames.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[allow(missing_docs)]
+pub struct ValueSummary<T> {
+    pub min: T,
+    pub max: T,
+    pub avg: f64,
+}
```

Comment:
this is just a copy-paste from above. Please, be sure that your code does not contain such repetitions.

Reply:
Ok, lets keep it!, thanks for the explanation.